internal/lookout/ui/src/services/ComputeResourcesService.ts

```diff
@@ -0,0 +1,111 @@
+import { load } from "js-yaml"
+
+// A simple interface that our PodSpec has.
+export interface JobYaml {
+  podSpec: {
+    containers: Container[]
+  }
+}
+
+export interface ContainerInfo {
+  command?: string[]
+  arguments?: string[]
+  cpu?: string[]
+  memory?: string[]
+  gpu?: string[]
+  "ephermal-storage"?: string[]
+}
+
+interface Container {
+  command: string[]
+  args: string[]
+  resources: {
+    limits: Limits
+  }
+}
+
+interface Limits {
+  cpu?: string
+  "nvidia.com/gpu"?: string
+  "ephemeral-storage"?: string
+  memory?: string
+}
+
+export function getContainerInfoFromYaml(jobYaml: string): ContainerInfo {
+  const yaml = load(jobYaml) as JobYaml
+  return {
+    command: getCommandFromJobYaml(yaml),
+    arguments: getCommandArgumentsFromJobYaml(yaml),
+    cpu: getCpuFromJobYaml(yaml),
+    memory: getMemoryFromJobYaml(yaml),
+    gpu: getGpuFromJobYaml(yaml),
+    "ephermal-storage": getStorageFromJobYaml(yaml),
+  }
+}
+
+function checkContainerFieldExists(yamlified: JobYaml): boolean {
+  return !yamlified.podSpec.containers
+}
+
+function getCommandFromJobYaml(yamlified: JobYaml): string[] | undefined {
+  if (checkContainerFieldExists(yamlified)) return
+  const stringArray: string[] = []
+  yamlified.podSpec.containers.map((val) => {
```

Comment:
Mentioned this already, but you shouldn't append values to an array in the map callback, rather return the values from the map itself. See the previous resolved comment for reference.

Reply:
Ah I see.